synfig-studio/src/synfigapp/actions/layerduplicate.cpp

```diff
@@ -418,21 +357,13 @@ LayerDuplicate::replace_valuenodes(const std::map<synfig::Layer::Handle,synfig::
 	for (const auto& layer_pair : cloned_layer_map) {
 
 		auto cloned_layer = layer_pair.second;
-
-		// disconnect_dynamic_param/connect_dynamic_param can change dynamic_param_list() while iterating
-		// which makes iter invalid, so we create a copy of dynamic_param_list() first
-		auto param_list = cloned_layer->dynamic_param_list();
-
-		for (auto iter=param_list.cbegin();iter!=param_list.cend();++iter)
-		{
-			for (const auto& vn_pair : cloned_valuenode_map) {
-				if (iter->second == vn_pair.first) {
-					cloned_layer->disconnect_dynamic_param(iter->first);
-					cloned_layer->connect_dynamic_param(iter->first, vn_pair.second);
-				} else {
-					do_replace_valuenodes(iter->second, vn_pair);
-				}
+		replace_value_nodes(cloned_layer,
+							[cloned_valuenode_map](ValueNode::LooseHandle vn) -> ValueNode::LooseHandle {
+			try {
```

Comment:
I think better to not use exceptions here:
```
    auto found = cloned_valuenode_map.find(vn);
    if (found == cloned_valuenode_map.end()) {
        return nullptr;
    }
    return found.second;
```

Reply:
Done